pkg/cri/sbserver/podsandbox/sandbox_stop.go

```diff
@@ -116,6 +122,13 @@ func (c *Controller) stopSandboxContainer(ctx context.Context, sandbox sandboxst
 	if err = task.Kill(ctx, syscall.SIGKILL); err != nil && !errdefs.IsNotFound(err) {
```

Comment:
Kill api just files singal to the processes in the group. It doesn't mean that the process has been stopped.
The process might be in D state.

Reply:
As you said, the kill API just sends the exit signal to the processes, it can't ensure the real exit, I think `task.Skill` does this for this reason, and can't get the exit status. 

Do you have any good advice on this? And if you think this needs improvement, I think we need to open a new issue to discuss how to fix it, the code is merged in `https://github.com/containerd/containerd/pull/7401`.

Reply:
Well. This comment is used to mention that if the process is still running, and `ExitedAt` has been updated, it is not correct. That is why I am confused about the new adding line `ExitedAt is the stop timestamp`.

Reply:
The status event change is already handled in https://github.com/containerd/containerd/blob/31f9d13f0cc3a7d60e009cb8e76ccb188d4a76d7/pkg/cri/sbserver/podsandbox/controller.go#L168. IMO, the new field should be updated in the function.

Reply:
As you said, modified to update in `task exit event`.

I think there is something wrong with the stop logic,https://github.com/containerd/containerd/blob/31f9d13f0cc3a7d60e009cb8e76ccb188d4a76d7/pkg/cri/sbserver/podsandbox/sandbox_stop.go#L116
In `state=sandboxstore.StateReady`, only task.Kill is called, no `task.Waite`, and the task exit event is sent.

Reply:
Thanks! 

The wait is here https://github.com/containerd/containerd/blob/31f9d13f0cc3a7d60e009cb8e76ccb188d4a76d7/pkg/cri/sbserver/podsandbox/sandbox_stop.go#L120. If the event is handled https://github.com/containerd/containerd/blob/31f9d13f0cc3a7d60e009cb8e76ccb188d4a76d7/pkg/cri/sbserver/podsandbox/controller.go#L174, the wait will return.

Reply:
Thanks for yours tips!